post.go

```diff
@@ -27,6 +27,23 @@ func (p *PostService) CreatePost(post *model.Post) error {
 	return nil
 }
 
+// DM sends a post as a direct message
+//
+// Minimum server version: 5.2
+func (p *PostService) DM(senderID, receiverID string, post *model.Post) (string, error) {
+	channel, appErr := p.api.GetDirectChannel(senderID, receiverID)
+	if appErr != nil {
+		return "", normalizeAppErr(appErr)
+	}
+	post.ChannelId = channel.Id
+	post.UserId = senderID
+	sentPost, appErr := p.api.CreatePost(post)
+	if appErr != nil {
+		return "", normalizeAppErr(appErr)
+	}
+	return sentPost.Id, nil
```

Comment:
This signature deviates from `CreatePost`, where the incoming `*model.Post` is considered mutable and overwritten with the created post (effectively assigning the `Id` back to the original post passed in.)

There was some discussion about this back in https://community-daily.mattermost.com/core/channels/plugin-api-rework-hackathon-2019, though I don't recall the particulars anymore. (@hanzei might remember?). I'm open to revisiting that discussion as necessary, but for now should we keep `DM` consistent with `CreatePost`?

Reply:
Sure. If the incoming *model.Post is mutated, we can follow the same pattern.